prql-compiler/src/sql/std.sql.prql

```diff
@@ -38,7 +38,8 @@ let read_parquet = source -> s"read_parquet({source})"
 let read_csv = source -> s"read_csv_auto({source})"
 
 let mul = l r -> null
-let div = l r -> null
+let div_i = l r -> s"ROUND({l} / {r} - 0.5)"
+let div_f = l r -> s"(CAST({l} AS FLOAT) / {r})"
```

Comment:
I am not sure that these are always correct.

- If `ROUND` is "Bankers' rounding", `ROUND(x - 0.5)` would not be `FROOR(x)`.
- If `l` is a type like FLOAT64, I think `CAST(l AS FLOAT)` is not correct. (See PEP 238 cited in this comment https://github.com/PRQL/prql/issues/2426#issuecomment-1554694777)

Reply:
> If ROUND is "Bankers' rounding", ROUND(x - 0.5) would not be FLOOR(x).

I used FLOOR before, but it was giving wrong answers for `-13 / 5`. I'll run tests on all databases locally now.

---

>If l is a type like FLOAT64, I think CAST(l AS FLOAT) is not correct.

Their argument is that it would not work for complex numbers. I don't think we should concern ourselves with that yet.

Reply:
> If ROUND is "Bankers' rounding", ROUND(x - 0.5) would not be FLOOR(x).

I used FLOOR before, but it was giving wrong answers for `-13 / 5`. I'll run tests on all databases locally now.

---

>If l is a type like FLOAT64, I think CAST(l AS FLOAT) is not correct.

Their argument is that it would not work for complex numbers. I don't think we should concern ourselves with that yet.

Reply:
Yes, this is correct — MySQL seems to have Decimal division rather than Float division...